File: app/src/main/java/com/pranayharjai7/myemotions/Database/DAO/EmotionDAO.java
package com.pranayharjai7.myemotions.Database.DAO;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.pranayharjai7.myemotions.Database.Emotion;

import java.util.List;

@Dao
public interface EmotionDAO {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertNewEmotion(Emotion emotion);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllEmotions(List<Emotion> emotions);

    @Query("SELECT * FROM Emotion")
    LiveData<List<Emotion>> getAllEmotions();

    @Query("SELECT * FROM Emotion WHERE userId = :userId")
    LiveData<List<Emotion>> getUserEmotions(String userId);

    @Query("DELETE FROM Emotion")
    void clearAllData();

    @Query("DELETE FROM Emotion WHERE userId = :userId")
    void clearUserData(String userId);
}

File: app/src/main/java/com/pranayharjai7/myemotions/data/local/room/EmotionDao.kt
package com.pranayharjai7.myemotions.data.local.room

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface EmotionDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertEmotion(emotion: EmotionRecordEntity)

    @Query("SELECT * FROM emotion_records WHERE userId = :userId ORDER BY timestamp DESC")
    fun getAllEmotions(userId: String): Flow<List<EmotionRecordEntity>>

    @Query("SELECT * FROM emotion_records WHERE synced = 0 AND userId = :userId ORDER BY timestamp ASC")
    suspend fun getUnsyncedEmotions(userId: String): List<EmotionRecordEntity>

    @Query("UPDATE emotion_records SET synced = 1 WHERE id = :id")
    suspend fun markAsSynced(id: String)
}
